src/pages/gl-integration-logic/Forms/IntegrationLogicDetailsForm.js

```diff
@@ -0,0 +1,229 @@
+import { Checkbox, FormControlLabel, Grid } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import { useForm } from 'src/hooks/form'
+import * as yup from 'yup'
+import FormShell from 'src/components/Shared/FormShell'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { ControlContext } from 'src/providers/ControlContext'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { DataSets } from 'src/resources/DataSets'
+import { ResourceLookup } from 'src/components/Shared/ResourceLookup'
+import { GeneralLedgerRepository } from 'src/repositories/GeneralLedgerRepository'
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+
+export default function IntegrationLogicDetailsForm({ ilId, recordId, labels, maxAccess, getGridData, window, editMode }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      ilId,
+      sign: '',
+      tagId: '',
+      integrationLevel: '',
+      masterSource: '',
+      postTypeId: '',
+      description: '',
+      isCostElement: false
+    },
+    enableReinitialize: true,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      sign: yup.string().required(),
+      integrationLevel: yup.string().required(),
+      masterSource: yup.string().required(),
+      postTypeId: yup.string().required()
+    }),
+    onSubmit: async obj => {
+      try {
+        const res2 = await fetchData()
```

Comment:
`    const res2 = await fetchData()
        const seqNo = res2.list.length > 0 ? res2.list[res2.list.length - 1]?.seqNo + 1 : 1`
        can we find better solution, since on each save we are fetching the whole data